src/main/java/com/cloudogu/wiki/WikiDispatcherServlet.java

```diff
@@ -78,6 +81,46 @@ protected void service(HttpServletRequest req, HttpServletResponse resp) throws
         }
     }
 
+    public List<Group> sortingWikisByGroups(Iterable<Wiki> wikis) {
+
+            /* getGroups gives a list with the groups including their repos.
+             * Therefore, the Wiki list (current order: 'repo 1 (group a), repo 2 (group a), ...')
+             * needs to be splitted into an order as 'group a (repo 1, ..), group b (repo x, ..), ..',
+             * that desired presentation in overviewRepos_en/de.html is possible.
+             */
+
+        String currentGroup = "";
+        List<Wiki> currentRepos = new ArrayList<Wiki>();
+        List<Group> groups = new ArrayList<Group>();
+
+        if(Iterables.size(wikis) != 0) {
+
+            for (Wiki wiki : wikis) { // wikis are already sorted by group name and alphabetically in ScmManager.getPotentialWikis
+
+                if (currentGroup != null && currentGroup.equals("")) { // first iteration -> currentGroup needs to get the current groupname
+                    currentGroup = wiki.getGroupName();
+                }
+                // after first iteration, groupname is the name of the group before
+                // if this equals with the current one, the current repo has the same group and can be added to currentRepos
+                if (currentGroup!=null && currentGroup.equals(wiki.getGroupName()) || currentGroup==wiki.getGroupName()) {
+                    currentRepos.add(wiki);
+                } else { //current repo does not have the same group as the repo before
+                    // -> the repos before can be added to groups, since all members of this group are found now
+                    if(currentGroup == null) currentGroup = "main";
+                    groups.add(new Group(currentGroup, currentRepos));
+                    currentRepos = new ArrayList<Wiki>(); // old currentRepos does not be needed anymore -> new initialisation
+                    currentRepos.add(wiki); //current repo has to be added
+                    // groupname of current repo has to be used for next iteration
+                    currentGroup = wiki.getGroupName();
+                }
+            }
+            if(currentGroup == null) currentGroup = "main";
+            groups.add(new Group(currentGroup, currentRepos)); //adding last group with last repos
+            return groups;
+        }
+        return null;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Return an empty collection instead of null. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://ecosystem.cloudogu.com/sonar/coding_rules#rule_key=squid%3AS1168)